vector/src/main/java/im/vector/app/features/home/room/list/home/release/ReleaseNotesViewModel.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.home.room.list.home.release
+
+import com.airbnb.mvrx.MavericksViewModelFactory
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import im.vector.app.core.di.MavericksAssistedViewModelFactory
+import im.vector.app.core.di.hiltMavericksViewModelFactory
+import im.vector.app.core.platform.VectorDummyViewState
+import im.vector.app.core.platform.VectorViewModel
+import timber.log.Timber
+
+class ReleaseNotesViewModel @AssistedInject constructor(
+        @Assisted initialState: VectorDummyViewState,
+) : VectorViewModel<VectorDummyViewState, ReleaseNotesAction, ReleaseNotesViewEvents>(initialState) {
+
+    @AssistedFactory
+    interface Factory : MavericksAssistedViewModelFactory<ReleaseNotesViewModel, VectorDummyViewState> {
+        override fun create(initialState: VectorDummyViewState): ReleaseNotesViewModel
+    }
+
+    companion object : MavericksViewModelFactory<ReleaseNotesViewModel, VectorDummyViewState> by hiltMavericksViewModelFactory()
+
+    private var selectedPageIndex = 0
```

Comment:
This should be moved to the ViewState (to create)

Reply:
I actually put it out from there, because there was a confusion

- sometimes page change is induced by button and in that case it comes from ViewModel
- sometimes page change is induced by user and it comes from Fragment

so, my initial attempt was to keep this index in State and update page indicator and button via view events, (since we can't react on each change, because some of them are induced by user (fragment)), but the problem is that view events and view state are both async and delivered by different streams, so there was a race problem